bin/PerlDDS/Run_Test.pm

```diff
@@ -464,7 +464,132 @@ sub wait_kill {
+      # doesn't have %p, then .PID is appended to the core file name.
+      my $uses_pid_file = "/proc/sys/kernel/core_uses_pid";
+      my $uses_pid = 0;
+      if (!open(my $uses_pid_fh, "<", "$uses_pid_file")) {
+          printf STDOUT "WARNING: Could not open $uses_pid_file\n";
+      } else {
+          $line = <$uses_pid_fh>;
+          if ($line ne "" || $line ne "\n") {
+              $uses_pid = $line;
+          }
+          close($uses_pid_fh);
+      }
+
+      my $exec_path = $process->Executable();
+
+      my $exec_name_idx = index($pattern, "%e");
+      if ($exec_name_idx != -1) {
+          my $exec_name = File::Basename::basename($exec_path);
+          # The core file name contains at most 15 characters from the executable name.
+          $exec_name = substr($exec_name, 0, 15);
+          substr($pattern, $exec_name_idx, 2) = $exec_name;
+      }
+
+      my $hname_idx = index($pattern, "%h");
+      if ($hname_idx != -1) {
+          substr($pattern, $hname_idx, 2) = Sys::Hostname::hostname();
+      }
+
+      my $pid_idx = index($pattern, "%p");
+      if ($pid_idx != -1) {
+          substr($pattern, $pid_idx, 2) = _getpid($process);
+      } elsif ($uses_pid != 0) {
+          $pattern = $pattern . "." . _getpid($process);
+      }
+
+      my $timestamp_idx = index($pattern, "%t");
+      my $core_file_path;
+      if ($timestamp_idx != -1) {
+          my $prefix = substr($pattern, 0, $timestamp_idx);
+          my $suffix_len = length($pattern) - $timestamp_idx - 2;
+          my $suffix = substr($pattern, $timestamp_idx + 2, $suffix_len);
+
+          # Get the core file that has latest timestamp.
+          opendir(my $dh, $path);
+          my @files = grep(/$prefix[0-9]+$suffix/, readdir($dh));
+          my $latest_timestamp;
+          my $chosen_core_file;
+          foreach my $file (@files) {
+              my $timestamp_len = length($file) - $timestamp_idx - $suffix_len;
+              my $timestamp = substr($file, $timestamp_idx, $timestamp_len);
+              if (!defined $latest_timestamp) {
+                  $latest_timestamp = $timestamp;
+                  $chosen_core_file = $file;
+              } elsif ($latest_timestamp < $timestamp) {
+                  $latest_timestamp = $timestamp;
+                  $chosen_core_file = $file;
+              }
+          }
+          closedir($dh);
+          if (defined $chosen_core_file) {
+              $core_file_path = $path . "/" . $chosen_core_file;
+          } else {
+              printf STDOUT "WARNING: Could not determine a core file with timestamp\n";
+              return $result;
+          }
+      } else {
+          $core_file_path = $path . "/" . $pattern;
+      }
+
+      if (!(-e $core_file_path)) {
+          printf STDOUT "WARNING: Core file $core_file_path not exist\n";
+          return $result;
+      }
+
+      # Print stack trace.
+      my $stack_trace;
+      if (system("gdb --version") != -1) {
```

Comment:
If we have both, but we built with clang, should we prefer to use lldb? I'm not sure there's an easy way to check for this so maybe it's fine to just try gdb first.

Reply:
Users can set the debugger they want now. Please see the PR to ACE.